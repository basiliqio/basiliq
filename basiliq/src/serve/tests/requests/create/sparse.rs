use super::*;

lazy_static::lazy_static! {
    pub static ref CREATE_BODY: serde_json::Value =
    json!({
        "data": json!({
            "type": "public__peoples",
            "attributes": json!({
                "first-name": "Francis",
                "last-name": "it's me",
                "twitter": "@myhandle",
                "gender": "M",
                "age": 22
            })
        })
    });
}

crate::run_test_request!(
    no_fields,
    Method::POST,
    "/public__peoples?fields[public__peoples]=",
    201,
    CREATE_BODY.clone()
);

crate::run_test_request!(
    one_fields,
    Method::POST,
    "/public__peoples?fields[public__peoples]=first-name",
    201,
    CREATE_BODY.clone()
);

crate::run_test_request!(
    multi_fields,
    Method::POST,
    "/public__peoples?fields[public__peoples]=first-name,gender",
    201,
    CREATE_BODY.clone()
);

crate::run_test_request!(
    unknown_fields,
    Method::POST,
    "/public__peoples?fields[public__peoples]=AAAAAAAAAAAAA",
    400,
    CREATE_BODY.clone()
);

crate::run_test_request!(
    unknown_type,
    Method::POST,
    "/public__peoples?fields[AAAAAAAAAAAAAA]=gender",
    400,
    CREATE_BODY.clone()
);
