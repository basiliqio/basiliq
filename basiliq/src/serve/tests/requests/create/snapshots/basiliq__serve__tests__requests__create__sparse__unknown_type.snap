---
source: basiliq/src/serve/tests/requests/create/sparse.rs
expression: res

---
{
  "errors": {
    "id": "[error id]",
    "status": 400,
    "title": "CibouletteUnknownType",
    "detail": "The json:api type `AAAAAAAAAAAAAA` is unknown."
  }
}
